tests/unit/package_managers/bundler/test_parser.py

```diff
@@ -180,3 +181,25 @@ def test_parse_gemlock_empty(
 
     assert f"Package {rooted_tmp_path.path.name} is bundled with version 2.5.10" in caplog.messages
     assert result == []
+
+
+@pytest.mark.parametrize(
+    "source",
+    [
+        "https://rubygems.org",
+        "https://dedicatedprivategemrepo.com",
+    ],
+)
+@mock.patch("cachi2.core.package_managers.bundler.parser.download_binary_file")
+def test_dependencies_could_be_downloaded(
+    mock_downloader: mock.MagicMock,
```

Comment:
Out of curiosity, what dunder methods do you need to type this as `MagicMock` instead of a regular `Mock`?

Reply:
I'd take a `MagicMock` anytime over an ordinary `Mock` :mage_man: 

In all seriousness, I think we tend to simply default to `Mock` in Cachi2, but probably not a big deal to keep it as `MagicMock`